chevah/brink
# [#2812] Add  support for creating a simple virtual env using paver.sh

Problem
=======

The current paver.sh script is tightly coupled to our infrastructure and our pavement / paver script.

We should find a way to decouple it so that it can be used to generate a simple python builder folder which could be reused by other projects.

Changes
=======

This is just an experimental code ... it works as it is used in chevah-keycert but is is brutal.

It gets the python binary from public location and install pip in it... just like virtualenv does... but virtualenv just creates some symlinks to local python installation.

it always gets latest version of pip over the Internet.

As a drive by I have update the code so that paver.sh will still work without a paver.conf


How to test
=========

reviewers: @dumol 

This is still WIP as I need to see how to make it work in a private LAN so that it will not touch the internet.

There are a few others changes needed by paver.sh so I try to make experimental versions for each code and then come with a refactored script:

* https://trac.chevah.com:10443/ticket/2808
* https://github.com/chevah/brink/pull/64
